network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/custom_dns/VpnCustomDnsActivity.kt

```diff
@@ -263,9 +258,18 @@ class VpnCustomDnsActivity : DuckDuckGoActivity() {
     }
 
     internal sealed class State {
-        // data class NeedApply(val value: Boolean) : State()
-        data class DefaultDns(val allowChange: Boolean, val blockMalware: Boolean) : State()
-        data class CustomDns(val dns: String?, val allowChange: Boolean, val applyEnabled: Boolean) : State()
+        data class DefaultDns(
+            val allowChange: Boolean,
+            val blockMalware: Boolean,
+            val allowBlockMalware: Boolean,
```

Comment:
I don't think we need this state that carries the kill-switch value. 
If we kill-switch the feature we should leave the UX untouched I think, we just stop blocking until we fix whatever broken.

I'd undo the chantes in the activity and view model

Reply:
I mainly added it as it might be confusing to the users if we don’t update the UI too.

Having that said, I don’t have strong opinions though. It is indeed enough from a kill switch standpoint to just not use the DNS even if the UI says otherwise.